spec/semaphore_spec.rb

```diff
@@ -148,6 +148,31 @@
       sleep 3.0
       expect(@redis.keys.count).to eq(original_key_size)
     end
+
+    it "does not leave an exists key without an expiration when an expiration is specified" do
+      original_key_size = @redis.keys.count
+      queue = Queue.new
+      threads = 2.times.collect do
+        Thread.new do
+          Redis::Semaphore.new(:my_semaphore, :redis => @redis, :expiration => 3).lock(5) do
+            sleep 1
+          end
+        end
+      end
+      sleep 4.0
+      @redis2 = Redis.new :db => 15
+      threads.each{|t| t.kill } #ensure signal step fails
```

Comment:
Missing space after #.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Pass `&:kill` as an argument to `each` instead of a block.